### PR TITLE
Extract anchor input component

### PR DESCRIPTION
**Pull request by nmattia**
This extracts the anchor input component (with input filter) from the authenticate page, which allows reuse in both the login and anchor prompt pages.

Also removes unnecessary anchor input CSS.

### CODE REVIEW

File: src/frontend/src/components/anchorInput.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { html } from "lit";
+import { createRef, ref } from "lit/directives/ref.js";
+import { TemplateRef } from "../utils/templateRef";
+
+/** A component for inputting an anchor number */
+export const mkAnchorInput = (
```
Comment (peterpeterparker) [inline src/frontend/src/components/anchorInput.ts:6]: `mk...` like `mock`?

File: src/frontend/src/flows/authenticate/index.ts [hunk 2]
```diff
@@ -46,8 +47,8 @@ const pageContent = (
   const onRecoverClick = () => useRecovery(connection, readUserNumber());
 
   const authorizeButton = createRef();
-  const userNumberInput = createRef();
   const registerButton = createRef();
+  const anchorInput = mkAnchorInput("userNumberInput", userNumber);
```
Comment (peterpeterparker) [inline src/frontend/src/flows/authenticate/index.ts:51]: for fun

```
const {template, refs: {userNumberInput}} = mkAnchorInput("userNumberInput", userNumber);
```

**Review by peterpeterparker — APPROVED** [review]
LGTM

Note: while reviewing this I just assumed that what's new is the component and new export of the template and refs - i.e. that the rest was just code moving around

File: src/frontend/src/components/anchorInput.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/components/anchorInput.ts:6]: ah no, like `make`! what do you suggest instead?

File: src/frontend/src/flows/authenticate/index.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/authenticate/index.ts:51]: I think `ts` would then complain about `template` being bound again...

**Comment by nmattia** [comment]
> the rest was just code moving around

Close! The `setInputFilter` was changed to use "lit" event binder, and the `flash-error` animation was removed, but in essence yes it's just code moved around

**Review by meodai — APPROVED** [review]
Looks good, cant wait for that!

File: src/frontend/src/components/anchorInput.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/components/anchorInput.ts:6]: oki doki, all good, thanks

Comment (nmattia) [reply src/frontend/src/components/anchorInput.ts:6]: clarified offline

**Pull request merged**